examples/tv-casting-app/APIs.md

```diff
@@ -0,0 +1,680 @@
+                contentLauncher.launchURL(
+		            new URL("https://YOUR_URL/id"), "displaystring", null);
+    		contentLauncherResponseFuture.thenApply(
+			(response) -> {
+				// handle the response
+				return null;
+			});
+}
+```
+
+On iOS, the commands can be sent as follows:
+
+```objectivec
+...
+if(endpoint.hasCluster(MTREndpointClusterTypeContentLauncher))
+{
+    let cluster: MTRContentLauncherCluster = endpoint.cluster(for: MTREndpointClusterTypeContentLauncher) as! MTRContentLauncherCluster
+    if(cluster.canLaunchURL())
+    {
+        cluster.launch(URL(string: "https://YOUR_URL/id")!) { launchResponse, err in
+            self.Log.info("invokeCommand completed with \(err)")
+            self.status = "ContentLauncher.LaunchURL completed!"
+
+            // consume the launchResponse
+        }
+    }
+    else
+    {
+        self.status = "Cannot launchURL on cluster"
+    }
+}
+else
+{
+    self.status = "No ContentLauncher cluster support found on the endpoint"
+}
+...
+```
+
+### Read Operations
+
+_{Complete Attribute Read examples:
+[Android](android/App/app/src/main/java/com/matter/casting/AttributeReadExample.java)
+| [iOS](darwin/TvCasting/TvCasting/MTRAttributeReadExampleViewModel.swift)}_
+
+The CastingClient may read an Attribute from the `contentAppEndpoint` on the
+CastingPlayer. It would ensure that the desired cluster / attribute are
+available for reading on the endpoint before trying to read it.
+
+On Android, an attribute can be read with implementation like the one below.
+
+```java
+// check if the Endpoint supports the Cluster to use
+if (contentAppEndpoint.get().hasCluster(MediaPlayback.class)) {
+    // get the Cluster to use
+    MediaPlayback mediaPlayback = contentAppEndpoint.get().getCluster(MediaPlayback.class);
+
+    Optional<Attribute<MediaPlayback.CurrentState>> currentStateAttribute = mediaPlayback.getCurrentState();
+    if(currentStateAttribute.isPresent() && currentStateAttribute.get().isAvailable())
+    {
+        CompletableFuture<MediaPlayback.CurrentState> currentStateValue = currentStateAttribute.get().read();
+        try {
+            Log.i(TAG, "Read Current State value: " + currentStateValue.get(1, TimeUnit.SECONDS));
+        } catch (ExecutionException | InterruptedException | TimeoutException e) {
+            Log.e(TAG, "Exception when reading CurrentState " + e);
+        }
+    }
+    else
+    {
+        Log.e(TAG, "Attribute unavailable on the selected endpoint");
+    }
+}
+else
+{
+    Log.e(TAG, "Required cluster not found on the selected endpoint");
+}
+```
+
+On iOS, an attribute can be read with implementation like the one below.
+
+```objectivec
+if(endpoint.hasCluster(MTREndpointClusterTypeContentLauncher))
+{
+    let cluster: MTRMediaPlaybackCluster = endpoint.cluster(for: MTREndpointClusterTypeMediaPlayback) as! MTRMediaPlaybackCluster
+    let currentStateAttribute: MTRAttribute<MTRCurrentState> = cluster.currentState
```

Comment:
It looks like in the iOS example MTRAttribute handles the optional. Can Android example leverage the same?

Reply:
I've added an isAvailable method and checked if it returns true here. Is that what you meant?

Reply:
I meant can't MTRAttribute encapsulate the logic of what optionals do today? I.e for each attribute definition, it will always return a MTRAttribute instance but has itself a isAvailable check in it that consumers can use to check if the value is available.

Reply:
This was already done, right? See line#582 for eg.